File: engine/sdk/py-kit-engine/kit_engine/identity.py
"""Identity management client."""

from __future__ import annotations

import requests


class IdentityClient:
    """REST client for kit identity endpoints."""

    def __init__(self, base_url: str, token: str | None = None):
        self._url = f"{base_url}/identity"
        self._token = token

    def _headers(self) -> dict[str, str]:
        if not self._token:
            return {}
        return {"Authorization": f"Bearer {self._token}"}

    def whoami(self) -> dict:
        resp = requests.get(self._url)
        resp.raise_for_status()
        return resp.json()

    def public_key(self) -> dict:
        return self.whoami()

    def verify(self, data: str, signature: str) -> dict:
        resp = requests.post(
            f"{self._url}/verify",
            json={"data": data, "signature": signature},
            headers=self._headers(),
        )
        resp.raise_for_status()
        return resp.json()

File: engine/sdk/ts-kit-engine/src/identity.ts
export interface PublicKeyInfo {
  id?: string;
  public_key: string;
  publicKey: string;
  fingerprint: string;
}

export interface VerifyResult { valid: boolean }

export class IdentityClient {
  constructor(private baseURL: string, private token?: string) {}

  async publicKey(): Promise<PublicKeyInfo> {
    const res = await fetch(`${this.baseURL}/identity`);
    if (!res.ok) throw new Error(`identity publicKey: ${res.status}`);
    const payload = (await res.json()) as Omit<PublicKeyInfo, "publicKey">;
    return { ...payload, publicKey: payload.public_key };
  }

  async verify(data: string, signature: string): Promise<VerifyResult> {
    const headers: Record<string, string> = { "Content-Type": "application/json" };
    if (this.token) headers["Authorization"] = `Bearer ${this.token}`;
    const res = await fetch(`${this.baseURL}/identity/verify`, {
      method: "POST",
      headers,
      body: JSON.stringify({ data, signature }),
    });
    if (!res.ok) throw new Error(`identity verify: ${res.status}`);
    return (await res.json()) as VerifyResult;
  }
}
